src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugeWave.cpp

```diff
@@ -0,0 +1,340 @@
+    const double /*t*/, const GaugeWave::IntermediateVars<DataType>& vars,
+    tmpl::list<gr::Tags::SqrtDetSpatialMetric<DataType>> /*meta*/) const
+    noexcept -> tuples::TaggedTuple<gr::Tags::SqrtDetSpatialMetric<DataType>> {
+  return {Scalar<DataType>{vars.sqrt_h}};
+}
+
+template <typename DataType>
+auto GaugeWave::variables(
+    const tnsr::I<DataType, GaugeWave::volume_dim, Frame::Inertial>& x,
+    const double /*t*/, const GaugeWave::IntermediateVars<DataType>& vars,
+    tmpl::list<gr::Tags::ExtrinsicCurvature<
+        GaugeWave::volume_dim, Frame::Inertial, DataType>> /*meta*/) const
+    noexcept -> tuples::TaggedTuple<gr::Tags::ExtrinsicCurvature<
+        GaugeWave::volume_dim, Frame::Inertial, DataType>> {
+  auto extrinsic_curvature = make_with_value<
+      tnsr::ii<DataType, GaugeWave::volume_dim, Frame::Inertial>>(x, 0.0);
+  get<0, 0>(extrinsic_curvature) = vars.dx_h_over_2_sqrt_h;
+  return extrinsic_curvature;
+}
+
+template <typename DataType>
+auto GaugeWave::variables(
+    const tnsr::I<DataType, GaugeWave::volume_dim, Frame::Inertial>& x,
+    const double /*t*/, const GaugeWave::IntermediateVars<DataType>& vars,
+    tmpl::list<gr::Tags::InverseSpatialMetric<
+        GaugeWave::volume_dim, Frame::Inertial, DataType>> /*meta*/) const
+    noexcept -> tuples::TaggedTuple<gr::Tags::InverseSpatialMetric<
+        GaugeWave::volume_dim, Frame::Inertial, DataType>> {
+  auto inverse_spatial_metric = make_with_value<
+      tnsr::II<DataType, GaugeWave::volume_dim, Frame::Inertial>>(x, 0.0);
+  get<0, 0>(inverse_spatial_metric) = 1.0 / vars.h;
+  for (size_t i = 1; i < GaugeWave::volume_dim; ++i) {
+    inverse_spatial_metric.get(i, i) = 1.0;
+  }
+  return inverse_spatial_metric;
+}
+
+bool operator==(const GaugeWave& lhs, const GaugeWave& rhs) noexcept {
+  return lhs.amplitude() == rhs.amplitude() and
+         lhs.wavelength() == rhs.wavelength();
+}
+
+bool operator!=(const GaugeWave& lhs, const GaugeWave& rhs) noexcept {
+  return not(lhs == rhs);
+}
+}  // namespace Solutions
+}  // namespace gr
+
+#define DIM(data) BOOST_PP_TUPLE_ELEM(0, data)
+#define DTYPE(data) BOOST_PP_TUPLE_ELEM(1, data)
+
+#define INSTANTIATE(_, data)                                                   \
+  template gr::Solutions::GaugeWave::IntermediateVars<DTYPE(data)>::           \
+      IntermediateVars(                                                        \
+          const double amplitude, const double wavelength,                     \
+          const tnsr::I<DTYPE(data), DIM(data), Frame::Inertial>& x,           \
+          const double t) noexcept;                                            \
+  template tuples::TaggedTuple<gr::Tags::Lapse<DTYPE(data)>>                   \
+  gr::Solutions::GaugeWave::variables(                                         \
+      const tnsr::I<DTYPE(data), DIM(data), Frame::Inertial>& /*x*/,           \
+      const double /*t*/,                                                      \
```

Comment:
Here in the template instantiations you don’t need the `const` :) They don’t hurt though, so I don’t request you change anything.

Reply:
Please keep this as is.